packages/nimble-components/src/table/index.ts

```diff
@@ -11,7 +22,116 @@ declare global {
 /**
  * A nimble-styled table.
  */
-export class Table extends FoundationElement {}
+export class Table<
+    TData extends TableData = TableData
+> extends FoundationElement {
+    @observable
+    public data: TData[] = [];
+
+    // TODO: Temporarily assume that the data in the table can be represented as strings.
+    @observable
+    public tableData: string[][] = [];
+
+    @observable
+    public columnHeaders: string[] = [];
```

Comment:
What's the motivation for adding this API now? Ultimately, do we expect this to be surfaced through the column API?

Reply:
Both `tableData` and `columnHeaders` is something that I see as being temporary. I just needed to expose the information somehow to be accessed within the template. I'll add a similar 'TODO' to `columnHeaders` as I put on `tableData`.